gems/aws-crt/native/src/api.h

```diff
@@ -42,6 +42,28 @@ AWS_CRT_API void aws_crt_reset_error(void);
 AWS_CRT_API struct aws_event_loop_group *aws_crt_event_loop_group_new(uint16_t max_threads);
 AWS_CRT_API void aws_crt_event_loop_group_release(struct aws_event_loop_group *elg);
 
+/* Auth */
+AWS_CRT_API struct aws_credentials *aws_crt_credentials_new(
+    const char *access_key_id,
+    const char *secret_access_key,
+    const char *session_token,
+    uint64_t expiration_timepoint_seconds);
+AWS_CRT_API void aws_crt_credentials_release(struct aws_credentials *credentials);
+AWS_CRT_API const char *aws_crt_credentials_get_access_key_id(struct aws_credentials *credentials);
+AWS_CRT_API const char *aws_crt_credentials_get_secret_access_key(struct aws_credentials *credentials);
+AWS_CRT_API const char *aws_crt_credentials_get_session_token(struct aws_credentials *credentials);
```

Comment:
oye ... big topic
So in our C APIs, we generally don't pass strings via `const char *` which requires the string to be null-terminated. Instead we use `struct aws_byte_cursor` which contains a pointer and a length, and the pointer **may not** be pointing at a null-terminated string. We did it because null-terminators are a common source of bugs in C code, and because we didn't want to require strings being passed in from other languages to be null-terminated.

Passing `const char *` as an argument *into* an aws_crt_ function seems fine, since FFI provides sugar for doing this. But we can't assume that an `aws_byte_cursor` coming *out* of C is null-terminated. We should probably bind the `aws_byte_cursor` class to ruby, for private use within the bindings. I think FFI provides some sugar to help bind simple structs like this.

Reply:
Ah!  Yeah.... that is an issue.  I'll look into create a binding for  aws_byte_cursor -> ruby string.

Reply:
I've added a Struct definition for aws_byte_cursor which does work but isn't quite as seamless as I'd like.